### PR TITLE
[SPARK-31411][UI] Show submitted time and duration in job details page

### PR DESCRIPTION
**Pull request by gengliangwang**
### What changes were proposed in this pull request?

Show submitted time and duration of a job in its details page

### Why are the changes needed?

When we check job details from the SQL execution page, it will be more convenient if we can get the submission time and duration from the job page, instead of finding the info from job list page.

### Does this PR introduce any user-facing change?

Yes. After changes, the job details page shows the submitted time and duration.

### How was this patch tested?

Manual check
image

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
retest this please

File: core/src/main/scala/org/apache/spark/ui/jobs/JobDataUtil.scala [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import org.apache.spark.status.api.v1.JobData
+import org.apache.spark.ui.UIUtils
+
+private[ui] object JobDataUtil {
+  def getDuration(jobData: JobData): Option[Long] = {
+    jobData.submissionTime.map { start =>
+        val end = jobData.completionTime.map(_.getTime()).getOrElse(System.currentTimeMillis())
+        end - start.getTime()
+    }
+  }
```
Comment (Ngone51) [inline core/src/main/scala/org/apache/spark/ui/jobs/JobDataUtil.scala:28]: nit: add a line?

**Review by Ngone51 — APPROVED** [review]

**Comment by srowen** [comment]
The only objection I have is that this is putting more info at the top of the page and pushing the details farther below the fold. Really this whole UX needs an overhaul to organize this better across the top and not just list info down the page. That's out of scope here, but how important is this to add?

**Comment by gengliangwang** [comment]
@srowen I understand your concerns. However, in the current UI, there is also duration info in the stage/SQL detail page as well. I think it is essential info for *details* page.

**Comment by sarutak** [comment]
retest this please.

**Comment by gengliangwang** [comment]
@srowen What do you think?

**Comment by srowen** [comment]
Do other details pages have something similar (sorry being lazy, I could look). I think consistency is a sufficient argument either way; do what other similar pages do

**Comment by gengliangwang** [comment]
@srowen Yes, stage page:
image

SQL page:
image

**Comment by sarutak** [comment]
LGTM. I agree with keeping consistency.

**Review by sarutak — APPROVED** [review]

**Comment by gengliangwang** [comment]
Merging to master.

**Pull request closed without merging**